src/J2N/Collections/Generic/OrderedDictionary.cs

```diff
@@ -2850,6 +2851,475 @@ public bool TryAdd([AllowNull] TAlternateKey key, TValue value)
+            if (!SpanAlternateLookup<TAlternateKeySpan>.IsCompatibleKey(this))
+            {
+                ThrowHelper.ThrowInvalidOperationException(ExceptionResource.InvalidOperation_IncompatibleComparer);
+            }
+
+            return new SpanAlternateLookup<TAlternateKeySpan>(this);
+        }
+
+        /// <summary>
+        /// Gets an instance of a type that may be used to perform operations on the current <see cref="OrderedDictionary{TKey, TValue}"/>
+        /// using a <see cref="ReadOnlySpan{T}"/> of type <typeparamref name="TAlternateKeySpan"/> as a key instead of a <typeparamref name="TKey"/>.
+        /// </summary>
+        /// <typeparam name="TAlternateKeySpan">The alternate <see cref="ReadOnlySpan{T}"/> type of a key for performing lookups.</typeparam>
+        /// <param name="lookup">The created lookup instance when the method returns true, or a default instance that should not be used if the method returns false.</param>
+        /// <returns><c>true</c> if a lookup could be created; otherwise, <c>false</c>.</returns>
+        /// <remarks>
+        /// The dictionary must be using a comparer that implements <see cref="ISpanAlternateEqualityComparer{TAlternateKeySpan, TKey}"/> with
+        /// a <see cref="ReadOnlySpan{T}"/> of type <typeparamref name="TAlternateKeySpan"/> and <typeparamref name="TKey"/>.
+        /// If it doesn't, the method will return false.
+        /// </remarks>
+        public bool TryGetSpanAlternateLookup<TAlternateKeySpan>(out SpanAlternateLookup<TAlternateKeySpan> lookup)
+        {
+            if (SpanAlternateLookup<TAlternateKeySpan>.IsCompatibleKey(this))
+            {
+                lookup = new SpanAlternateLookup<TAlternateKeySpan>(this);
+                return true;
+            }
+
+            lookup = default;
+            return false;
+        }
+
+        /// <summary>
+        /// Provides a type that may be used to perform operations on an <see cref="OrderedDictionary{TKey, TValue}"/>
+        /// using a <see cref="ReadOnlySpan{T}"/> of type <typeparamref name="TAlternateKeySpan"/> as a key instead of a <typeparamref name="TKey"/>.
+        /// </summary>
+        /// <typeparam name="TAlternateKeySpan">The alternate <see cref="ReadOnlySpan{T}"/> type of a key for performing lookups.</typeparam>
+        public readonly struct SpanAlternateLookup<TAlternateKeySpan>
+        {
+            /// <summary>Initialize the instance. The dictionary must have already been verified to have a compatible comparer.</summary>
+            internal SpanAlternateLookup(OrderedDictionary<TKey, TValue> dictionary)
+            {
+                Debug.Assert(dictionary is not null);
+                Debug.Assert(IsCompatibleKey(dictionary!)); // [!]: asserted above
+                Dictionary = dictionary!; // [!]: asserted above
+            }
+
+            /// <summary>Gets the <see cref="Dictionary{TKey, TValue}"/> against which this instance performs operations.</summary>
```

Comment:
Comment mentions `Dictionary{TKey, TValue}` but should be `OrderedDictionary{TKey, TValue}`